src/components/commander.ts

```diff
@@ -15,16 +18,29 @@ export class LaTeXCommandTreeView {
     }
 }
 
-class LaTeXCommander implements vscode.TreeDataProvider<LaTeXCommand> {
-
-    private readonly commands: LaTeXCommand[] = []
+export class LaTeXCommanderProvider implements vscode.TreeDataProvider<LaTeXCommand> {
+    private readonly extension: Extension
+    private readonly _onDidChangeTreeData: vscode.EventEmitter<LaTeXCommand | undefined> = new vscode.EventEmitter<LaTeXCommand | undefined>()
+    readonly onDidChangeTreeData: vscode.Event<LaTeXCommand | undefined>
```

Comment:
Move  `_onDidChangeTreeData` and `onDidChangeTreeData` to `class LaTeXCommandTreeView`

Reply:
According to https://code.visualstudio.com/api/references/vscode-api#TreeDataProvider%3CT%3E, the event `onDidChangeTreeData` must be defined in the `TreeDataProvider` not in the `TreeView` and the documentation says

>  This will trigger the view to update the changed element/root and its children recursively (if shown).

I could define the `LatexCommanderProvider` directly in the `LaTeXCommandTreeView` and chain the two updates but it would not be coherent with what we do in `structure.ts` and would not really follow the logic of VS Code's API. I think it is better to follow as much as possible the same patterns in all the components.

Even though the `LatexCommanderProvider` is not really needed at the `Extension` level per se, could you expand a bit on the motivations behind 

> `class Extention` should not have providers directly

Reply:
> it would not be coherent with what we do in structure.ts

I don't think what we do in `structure.ts` follows good practice. We had better avoid exposing implementation details. And, `SectionNodeProvider` is an implementation detail. Other components should access it through `StructureTreeView`.

In other words, it is good we have two or more classes, `SectionNodeProvider`, `StructureTreeView`, and others, to implement the structure tree feature. However, we don't have to expose both classes. Hiding implementation details increases the maintainability of the codebase. Splitting a thing into small components is good. Exposing all of them as API is not good.

We can say the same thing with `LaTeXCommandTreeView` and `LaTeXCommanderProvider`. It is good to have multiple classes to implement something. But, we don't have to expose both.

Reply:
> According to https://code.visualstudio.com/api/references/vscode-api#TreeDataProvider%3CT%3E, the event onDidChangeTreeData must be defined in the TreeDataProvider not in the TreeView and the documentation says

Sorry. We don't have to move `onDidChangeTreeData`.

Reply:
I have refactored the implementation to stop exposing `LaTeXCommanderProvider`. There was a mix between `LaTeXCommand` and `LaTeXCommander`. Now, we use `LaTeXCommander` everywhere.